--- a/target/linux/ath79/base-files/etc/board.d/01_leds
+++ b/target/linux/ath79/base-files/etc/board.d/01_leds
@@ -150,6 +150,7 @@ tplink,archer-d50-v1)
 	ucidef_set_led_switch "wan_data" "WAN Data" "tp-link:white:internet" "switch0" "0x02" "" "tx rx"
 	ucidef_set_led_switch "wan_link" "WAN Link" "tp-link:white:wan" "switch0" "0x02" "" "link"
 	;;
+tplink,cpe510-v1|\
 tplink,cpe220-v2|\
 tplink,cpe220-v3)
 	ucidef_set_led_netdev "lan0" "LAN0" "tp-link:green:lan0" "eth1"
--- a/target/linux/ath79/base-files/etc/board.d/02_network
+++ b/target/linux/ath79/base-files/etc/board.d/02_network
@@ -102,6 +102,7 @@ ath79_setup_interfaces()
 		;;
 	comfast,cf-e110n-v2|\
 	comfast,cf-e120a-v3|\
+	tplink,cpe510-v1|\
 	tplink,cpe220-v2|\
 	tplink,cpe220-v3|\
 	ubnt,nanostation-m|\
--- a/target/linux/ath79/base-files/etc/board.d/03_gpio_switches
+++ b/target/linux/ath79/base-files/etc/board.d/03_gpio_switches
@@ -27,6 +27,7 @@ tplink,archer-c25-v1)
 	ucidef_add_gpio_switch "led_control" "LED control" "21" "0"
 	ucidef_add_gpio_switch "led_reset" "LED reset" "19" "1"
 	;;
+tplink,cpe510-v1|\
 tplink,cpe220-v2|\
 tplink,cpe220-v3)
 	ucidef_add_gpio_switch "poe_passthrough" "PoE Passthrough" "20"
--- /dev/null
+++ b/target/linux/ath79/dts/ar9344_tplink_cpe510-v1.dts
@@ -0,0 +1,13 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include "ar9344_tplink_cpe_2port.dtsi"
+
+/ {
+	compatible = "tplink,cpe510-v1", "qca,ar9344";
+	model = "TP-Link CPE510 v1";
+};
+
+&led_link4 {
+	gpios = <&gpio 16 GPIO_ACTIVE_LOW>;
+};
--- a/target/linux/ath79/image/generic-tp-link.mk
+++ b/target/linux/ath79/image/generic-tp-link.mk
@@ -195,6 +195,22 @@ define Device/tplink_cpe220-v3
 endef
 TARGET_DEVICES += tplink_cpe220-v3
 
+define Device/tplink_cpe510-v1
+  $(Device/tplink-safeloader)
+  ATH_SOC := ar9344
+  IMAGE_SIZE := 7680k
+  DEVICE_TITLE := TP-Link CPE510 v1
+  DEVICE_PACKAGES := rssileds
+  TPLINK_BOARD_ID := CPE510
+  LOADER_TYPE := elf
+  LOADER_FLASH_OFFS := 0x43000 
+  COMPILE := loader-$(1).elf 
+  COMPILE/loader-$(1).elf := loader-okli-compile 
+  KERNEL := kernel-bin | append-dtb | lzma | uImage lzma -M 0x4f4b4c49 | loader-okli $(1) 12288
+  SUPPORTED_DEVICES += cpe510
+endef
+TARGET_DEVICES += tplink_cpe510-v1
+
 define Device/tplink_cpe510-v2
   $(Device/tplink-safeloader)
   ATH_SOC := ar9344
