Index: openwrt/target/linux/ar71xx/base-files/etc/board.d/01_leds
===================================================================
--- openwrt.orig/target/linux/ar71xx/base-files/etc/board.d/01_leds
+++ openwrt/target/linux/ar71xx/base-files/etc/board.d/01_leds
@@ -422,8 +422,8 @@ gl-mifi)
 	;;
 gl-ar150)
 	ucidef_set_led_wlan "wlan" "WLAN" "$board:orange:wlan" "phy0tpt"
-	ucidef_set_led_netdev "wan" "WAN" "$board:green:wan" "eth0"
-	ucidef_set_led_netdev "lan" "LAN" "$board:green:lan" "eth1"
+	ucidef_set_led_netdev "wan" "WAN" "$board:green:lan" "eth0"
+	ucidef_set_led_netdev "lan" "LAN" "$board:green:wan" "eth1"
 	;;
 gl-ar300)
 	ucidef_set_led_wlan "wlan" "WLAN" "$board:wlan" "phy0tpt"
