Index: openwrt/target/linux/ath79/image/generic-tp-link.mk
===================================================================
--- openwrt.orig/target/linux/ath79/image/generic-tp-link.mk
+++ openwrt/target/linux/ath79/image/generic-tp-link.mk
@@ -175,6 +175,7 @@ define Device/tplink_cpe220-v3
   DEVICE_PACKAGES := rssileds
   TPLINK_BOARD_ID := CPE220V3
   LOADER_TYPE := elf
+  SUPPORTED_DEVICES += cpe220-v3
 endef
 TARGET_DEVICES += tplink_cpe220-v3
 
