Index: openwrt/target/linux/ar71xx/base-files/etc/board.d/01_leds
===================================================================
--- openwrt.orig/target/linux/ar71xx/base-files/etc/board.d/01_leds
+++ openwrt/target/linux/ar71xx/base-files/etc/board.d/01_leds
@@ -680,7 +680,8 @@ rb-2011uias-2hnd)
 	ucidef_set_led_switch "eth9" "ETH9" "rb:green:eth9" "switch1" "0x04"
 	ucidef_set_led_switch "eth10" "ETH10" "rb:green:eth10" "switch1" "0x02"
 	;;
-rb-lhg-5nd)
+rb-lhg-5nd|\
+rb-lhg-5hpnd-xl)
 	ucidef_set_led_netdev "lan" "LAN" "rb:green:eth" "eth0"
 	ucidef_set_rssimon "wlan0" "200000" "1"
 	ucidef_set_led_rssi "rssilow" "RSSILOW" "rb:green:rssi0" "wlan0" "1" "100" "0" "13"
Index: openwrt/target/linux/ar71xx/base-files/etc/board.d/02_network
===================================================================
--- openwrt.orig/target/linux/ar71xx/base-files/etc/board.d/02_network
+++ openwrt/target/linux/ar71xx/base-files/etc/board.d/02_network
@@ -113,6 +113,7 @@ ar71xx_setup_interfaces()
 	rb-912uag-5hpnd|\
 	rb-921gs-5hpacd-r2|\
 	rb-lhg-5nd|\
+	rb-lhg-5hpnd-xl|\
 	rb-mapl-2nd|\
 	rb-sxt2n|\
 	rb-sxt5n|\
Index: openwrt/target/linux/ar71xx/base-files/etc/diag.sh
===================================================================
--- openwrt.orig/target/linux/ar71xx/base-files/etc/diag.sh
+++ openwrt/target/linux/ar71xx/base-files/etc/diag.sh
@@ -376,6 +376,7 @@ get_status_led() {
 	rb-952ui-5ac2nd|\
 	rb-962uigs-5hact2hnt|\
 	rb-lhg-5nd|\
+	rb-lhg-5hpnd-xl|\
 	rb-map-2nd|\
 	rb-mapl-2nd|\
 	rb-wap-2nd)
Index: openwrt/target/linux/ar71xx/base-files/lib/ar71xx.sh
===================================================================
--- openwrt.orig/target/linux/ar71xx/base-files/lib/ar71xx.sh
+++ openwrt/target/linux/ar71xx/base-files/lib/ar71xx.sh
@@ -1118,6 +1118,9 @@ ar71xx_board_detect() {
 	*"RouterBOARD LHG 5nD")
 		name="rb-lhg-5nd"
 		;;
+	*"RouterBOARD LHG 5HPnD-XL")
+		name="rb-lhg-5hpnd-xl"
+		;;
 	*"RouterBOARD mAP 2nD")
 		name="rb-map-2nd"
 		;;
Index: openwrt/target/linux/ar71xx/base-files/lib/upgrade/platform.sh
===================================================================
--- openwrt.orig/target/linux/ar71xx/base-files/lib/upgrade/platform.sh
+++ openwrt/target/linux/ar71xx/base-files/lib/upgrade/platform.sh
@@ -720,6 +720,7 @@ platform_check_image() {
 	rb-952ui-5ac2nd|\
 	rb-962uigs-5hact2hnt|\
 	rb-lhg-5nd|\
+	rb-lhg-5hpnd-xl|\
 	rb-map-2nd|\
 	rb-mapl-2nd|\
 	rb-wap-2nd|\
@@ -746,6 +747,7 @@ platform_pre_upgrade() {
 	rb-952ui-5ac2nd|\
 	rb-962uigs-5hact2hnt|\
 	rb-lhg-5nd|\
+	rb-lhg-5hpnd-xl|\
 	rb-map-2nd|\
 	rb-mapl-2nd|\
 	rb-wap-2nd|\
Index: openwrt/target/linux/ar71xx/image/mikrotik.mk
===================================================================
--- openwrt.orig/target/linux/ar71xx/image/mikrotik.mk
+++ openwrt/target/linux/ar71xx/image/mikrotik.mk
@@ -40,7 +40,7 @@ define Device/rb-nor-flash-16M
   LOADER_TYPE := elf
   KERNEL_INSTALL := 1
   KERNEL := kernel-bin | lzma | loader-kernel
-  SUPPORTED_DEVICES := rb-750-r2 rb-750up-r2 rb-750p-pbr2 rb-911-2hn rb-911-5hn rb-941-2nd rb-951ui-2nd rb-952ui-5ac2nd rb-962uigs-5hact2hnt rb-lhg-5nd rb-map-2nd rb-mapl-2nd rb-wap-2nd
+  SUPPORTED_DEVICES := rb-750-r2 rb-750up-r2 rb-750p-pbr2 rb-911-2hn rb-911-5hn rb-941-2nd rb-951ui-2nd rb-952ui-5ac2nd rb-962uigs-5hact2hnt rb-lhg-5nd rb-lhg-5hpnd-xl rb-map-2nd rb-mapl-2nd rb-wap-2nd
   IMAGE/sysupgrade.bin := append-kernel | kernel2minor -s 1024 -e | pad-to $$$$(BLOCKSIZE) | \
 	append-rootfs | pad-rootfs | append-metadata | check-size $$$$(IMAGE_SIZE)
 endef
