Index: openwrt/target/linux/ar71xx/base-files/etc/board.d/01_leds
===================================================================
--- openwrt.orig/target/linux/ar71xx/base-files/etc/board.d/01_leds
+++ openwrt/target/linux/ar71xx/base-files/etc/board.d/01_leds
@@ -693,7 +693,8 @@ rb-2011uias-2hnd-r2)
 	ucidef_set_led_switch "eth10" "ETH10" "rb:green:eth10" "switch1" "0x02"
 	;;
 rb-lhg-5nd|\
-rb-lhg-5hpnd-xl)
+rb-lhg-5hpnd-xl|\
+rb-ldf-5nd)
 	ucidef_set_led_netdev "lan" "LAN" "rb:green:eth" "eth0"
 	ucidef_set_rssimon "wlan0" "200000" "1"
 	ucidef_set_led_rssi "rssilow" "RSSILOW" "rb:green:rssi0" "wlan0" "1" "100" "0" "13"
Index: openwrt/target/linux/ar71xx/base-files/etc/board.d/02_network
===================================================================
--- openwrt.orig/target/linux/ar71xx/base-files/etc/board.d/02_network
+++ openwrt/target/linux/ar71xx/base-files/etc/board.d/02_network
@@ -115,6 +115,7 @@ ar71xx_setup_interfaces()
 	rb-922uags-5hpacd|\
 	rb-lhg-5nd|\
 	rb-lhg-5hpnd-xl|\
+	rb-ldf-5nd|\
 	rb-mapl-2nd|\
 	rb-sxt2n|\
 	rb-sxt-2nd-r3|\
Index: openwrt/target/linux/ar71xx/base-files/etc/diag.sh
===================================================================
--- openwrt.orig/target/linux/ar71xx/base-files/etc/diag.sh
+++ openwrt/target/linux/ar71xx/base-files/etc/diag.sh
@@ -379,6 +379,7 @@ get_status_led() {
 	rb-962uigs-5hact2hnt|\
 	rb-lhg-5nd|\
 	rb-lhg-5hpnd-xl|\
+	rb-ldf-5nd|\
 	rb-map-2nd|\
 	rb-mapl-2nd|\
 	rb-sxt-2nd-r3|\
Index: openwrt/target/linux/ar71xx/base-files/lib/ar71xx.sh
===================================================================
--- openwrt.orig/target/linux/ar71xx/base-files/lib/ar71xx.sh
+++ openwrt/target/linux/ar71xx/base-files/lib/ar71xx.sh
@@ -528,6 +528,12 @@ mikrotik_board_detect() {
 	*"LHG 5HPnD-XL")
 		name="rb-lhg-5hpnd-xl"
 		;;
+	*"LDF-5nD")
+		name="rb-ldf-5nd"
+		;;
+	*"RBLDF-5nD")
+		name="rb-ldf-5nd"
+		;;
 	*"mAP 2nD"|\
 	*"mAP2nD")
 		name="rb-map-2nd"
Index: openwrt/target/linux/ar71xx/base-files/lib/upgrade/platform.sh
===================================================================
--- openwrt.orig/target/linux/ar71xx/base-files/lib/upgrade/platform.sh
+++ openwrt/target/linux/ar71xx/base-files/lib/upgrade/platform.sh
@@ -729,6 +729,7 @@ platform_check_image() {
 	rb-962uigs-5hact2hnt|\
 	rb-lhg-5nd|\
 	rb-lhg-5hpnd-xl|\
+	rb-ldf-5nd|\
 	rb-map-2nd|\
 	rb-mapl-2nd|\
 	rb-sxt-2nd-r3|\
@@ -759,6 +760,7 @@ platform_pre_upgrade() {
 	rb-962uigs-5hact2hnt|\
 	rb-lhg-5nd|\
 	rb-lhg-5hpnd-xl|\
+	rb-ldf-5nd|\
 	rb-map-2nd|\
 	rb-mapl-2nd|\
 	rb-sxt-2nd-r3|\
Index: openwrt/target/linux/ar71xx/image/mikrotik.mk
===================================================================
--- openwrt.orig/target/linux/ar71xx/image/mikrotik.mk
+++ openwrt/target/linux/ar71xx/image/mikrotik.mk
@@ -46,7 +46,7 @@ define Device/rb-nor-flash-16M
   DEVICE_PACKAGES := rbcfg rssileds -nand-utils kmod-ledtrig-gpio
   IMAGE_SIZE := 16000k
   KERNEL_INSTALL := 1
-  SUPPORTED_DEVICES := rb-750-r2 rb-750up-r2 rb-750p-pbr2 rb-911-2hn rb-911-5hn rb-931-2nd rb-941-2nd rb-951ui-2nd rb-952ui-5ac2nd rb-962uigs-5hact2hnt rb-lhg-5nd rb-lhg-5hpnd-xl rb-map-2nd rb-mapl-2nd rb-wap-2nd rb-wapr-2nd rb-sxt-2nd-r3
+  SUPPORTED_DEVICES := rb-750-r2 rb-750up-r2 rb-750p-pbr2 rb-911-2hn rb-911-5hn rb-931-2nd rb-941-2nd rb-951ui-2nd rb-952ui-5ac2nd rb-962uigs-5hact2hnt rb-lhg-5nd rb-lhg-5hpnd-xl rb-ldf-5nd rb-map-2nd rb-mapl-2nd rb-wap-2nd rb-wapr-2nd rb-sxt-2nd-r3
   IMAGE/sysupgrade.bin := append-kernel | kernel2minor -s 1024 -e | pad-to $$$$(BLOCKSIZE) | \
 	append-rootfs | pad-rootfs | append-metadata | check-size $$$$(IMAGE_SIZE)
 endef
