Index: openwrt/target/linux/ar71xx/base-files/lib/ar71xx.sh
===================================================================
--- openwrt.orig/target/linux/ar71xx/base-files/lib/ar71xx.sh
+++ openwrt/target/linux/ar71xx/base-files/lib/ar71xx.sh
@@ -519,6 +519,9 @@ mikrotik_board_detect() {
 	*"952Ui-5ac2nD")
 		name="rb-952ui-5ac2nd"
 		;;
+	*"RB952Ui-5ac2nD")
+		name="rb-952ui-5ac2nd"
+		;;
 	*"962UiGS-5HacT2HnT")
 		name="rb-962uigs-5hact2hnt"
 		;;
