Index: openwrt/target/linux/ar71xx/base-files/lib/ar71xx.sh
===================================================================
--- openwrt.orig/target/linux/ar71xx/base-files/lib/ar71xx.sh
+++ openwrt/target/linux/ar71xx/base-files/lib/ar71xx.sh
@@ -1106,6 +1106,9 @@ ar71xx_board_detect() {
 	*"RouterBOARD 952Ui-5ac2nD")
 		name="rb-952ui-5ac2nd"
 		;;
+	*"RouterBOARD RB952Ui-5ac2nD")
+		name="rb-952ui-5ac2nd"
+		;;
 	*"RouterBOARD 962UiGS-5HacT2HnT")
 		name="rb-962uigs-5hact2hnt"
 		;;
