Index: openwrt/target/linux/ar71xx/base-files/lib/ar71xx.sh
===================================================================
--- openwrt.orig/target/linux/ar71xx/base-files/lib/ar71xx.sh
+++ openwrt/target/linux/ar71xx/base-files/lib/ar71xx.sh
@@ -1163,6 +1163,15 @@ ar71xx_board_detect() {
 	*"RouterBOARD SXT Lite5")
 		name="rb-sxt5n"
 		;;
+	*"RouterBOARD SXTsq 5nD")
+		name="rb-lhg-5nd"
+		;;
+	*"RouterBOARD SXTsq 2nD")
+		name="rb-lhg-5nd"
+		;;
+	*"RouterBOARD SXTsq 5HPnD")
+		name="rb-lhg-5hpnd"
+		;;
 	*"RouterBOARD wAP 2nD r2")
 		name="rb-wap-2nd"
 		;;
