Index: openwrt/target/linux/ar71xx/base-files/lib/ar71xx.sh
===================================================================
--- openwrt.orig/target/linux/ar71xx/base-files/lib/ar71xx.sh
+++ openwrt/target/linux/ar71xx/base-files/lib/ar71xx.sh
@@ -577,6 +577,19 @@ mikrotik_board_detect() {
 	*"SXT Lite5")
 		name="rb-sxt5n"
 		;;
+	*"SXTsq 5nD")
+		name="rb-lhg-5nd"
+		;;
+	*"SXTsq 2nD")
+		name="rb-lhg-5nd"
+		;;
+	*"RBSXTsq2nD")
+		name="rb-lhg-5nd"
+		;;
+	*"SXTsq 5HPnD"|\
+	*"SXTsq5HPnD")
+		name="rb-lhg-5hpnd"
+		;;
 	*"wAP 2nD r2")
 		name="rb-wap-2nd"
 		;;
