From bb336671fff613b812b308c7eeeeda38608bda9d Mon Sep 17 00:00:00 2001
From: Jaymin Patel <Jaymin.Patel@Sophos.com>
Date: Mon, 30 Jul 2018 15:06:14 +0530
Subject: [PATCH] fix compilation failure when luasrcdiet is being copied to
 non existing directory

Signed-off-by: Jaymin Patel <jem.patel@gmail.com>
Index: openwrt/feeds/luci/modules/luci-base/Makefile
===================================================================
--- openwrt.orig/feeds/luci/modules/luci-base/Makefile
+++ openwrt/feeds/luci/modules/luci-base/Makefile
@@ -41,6 +41,7 @@ endef
 
 define Host/Install
 	$(INSTALL_DIR) $(1)/bin
+	$(INSTALL_DIR) $(1)/lib/lua/5.1
 	$(INSTALL_BIN) src/po2lmo $(1)/bin/po2lmo
 	$(INSTALL_BIN) $(HOST_BUILD_DIR)/bin/luasrcdiet $(1)/bin/luasrcdiet
 	$(CP) $(HOST_BUILD_DIR)/luasrcdiet $(1)/lib/lua/5.1/
