
--- a/target/linux/ar71xx/base-files/etc/diag.sh
+++ b/target/linux/ar71xx/base-files/etc/diag.sh
@@ -155,6 +155,7 @@ get_status_led() {
 	cf-e385ac)
 		status_led="$board:blue:wlan2g"
 		;;
+	cpe210|\
 	cpe510)
 		status_led="tp-link:green:link4"
 		;;
--- a/target/linux/ar71xx/base-files/etc/diag.sh
+++ b/target/linux/ar71xx/base-files/etc/diag.sh
@@ -155,6 +155,7 @@ get_status_led() {
 	cf-e385ac)
 		status_led="$board:blue:wlan2g"
 		;;
+	cpe210-v2|\
 	cpe210|\
 	cpe510)
 		status_led="tp-link:green:link4"
