--- a/package/base-files/files/etc/banner
+++ b/package/base-files/files/etc/banner
@@ -1,8 +1,16 @@
-  _______                     ________        __
- |       |.-----.-----.-----.|  |  |  |.----.|  |_
- |   -   ||  _  |  -__|     ||  |  |  ||   _||   _|
- |_______||   __|_____|__|__||________||__|  |____|
-          |__| W I R E L E S S   F R E E D O M
- -----------------------------------------------------
- %D %V, %C
- -----------------------------------------------------
+              _____  ______ _____  _   _ 
+        /\   |  __ \|  ____|  __ \| \ | |TM
+       /  \  | |__) | |__  | |  | |  \| |
+      / /\ \ |  _  /|  __| | |  | | . ` |
+     / ____ \| | \ \| |____| |__| | |\  |
+    /_/    \_\_|  \_\______|_____/|_| \_| 
+    AMATEUR  RADIO EMERGENCY DATA NETWORK
+-----------------------------------------------
+ * 1 Battery          Connect all devices
+ * 2 POE injectors    Upgrade firmware to AREDN
+ * 3 cat5 cables      Setup with your callsign
+ * 1 UBNT NanoStation Point the Antenna
+ * 1 ipCam            Welcome to the Mesh!
+-----------------------------------------------
+  %V, %C
+ ----------------------------------------------
