--- a/target/linux/ipq40xx/base-files/etc/board.d/02_network
+++ b/target/linux/ipq40xx/base-files/etc/board.d/02_network
@@ -33,7 +33,9 @@ ipq40xx_setup_interfaces()
 	netgear,ex6100v2|\
 	netgear,ex6150v2|\
 	zyxel,wre6606)
-		ucidef_set_interface_lan "eth0"
+		ucidef_set_interfaces_lan_wan "eth0" "eth0.1"
+		ucidef_set_interface "dtdlink" device "eth0.2" protocol "static"
+		ucidef_set_interface "wifi" device "wlan0" protocol "static"
 		;;
 	aruba,ap-303h)
 		ucidef_set_interfaces_lan_wan "eth0" "eth1"
@@ -51,9 +53,9 @@ ipq40xx_setup_interfaces()
 	asus,rt-ac58u|\
 	mikrotik,hap-ac2|\
 	zyxel,nbg6617)
-		ucidef_set_interfaces_lan_wan "eth0" "eth1"
+		ucidef_set_interface_wan "eth1"
 		ucidef_add_switch "switch0" \
-			"0u@eth0" "1:lan:4" "2:lan:3" "3:lan:2" "4:lan:1"
+			"0u@eth0" "1:lan" "2:lan" "3:lan" "4t:dummy" "4t:dtdlink"
 		;;
 	avm,fritzbox-4040|\
 	linksys,ea6350v3|\
