--- a/target/linux/ath79/tiny/base-files/etc/board.d/02_network
+++ b/target/linux/ath79/tiny/base-files/etc/board.d/02_network
@@ -57,7 +57,9 @@
 	ubnt,nanostation-loco-m|\
 	ubnt,powerbridge-m|\
 	ubnt,rocket-m)
-		ucidef_set_interface_lan "eth0"
+		ucidef_set_interfaces_lan_wan "eth0" "eth0.1"
+		ucidef_set_interface "dtdlink" device "eth0.2" protocol "static"
+		ucidef_set_interface "wifi" device "wlan0" protocol "static"
 		;;
 	engenius,enh202-v1)
 		ucidef_set_interface_lan "eth0"
