Index: openwrt/package/base-files/files/lib/functions/uci-defaults.sh
===================================================================
--- openwrt.orig/package/base-files/files/lib/functions/uci-defaults.sh
+++ openwrt/package/base-files/files/lib/functions/uci-defaults.sh
@@ -153,7 +153,7 @@ _ucidef_finish_switch_roles() {
 
 				if [ ${need_tag:-0} -eq 1 -o ${want_untag:-0} -ne 1 ]; then
 					num="${num}t"
-					device="${device}.${index}"
+					device="${device}.$((index - 1))"
 				fi
 
 				json_select roles
Index: openwrt/package/base-files/files/bin/config_generate
===================================================================
--- openwrt.orig/package/base-files/files/bin/config_generate
+++ openwrt/package/base-files/files/bin/config_generate
@@ -165,10 +165,11 @@ generate_switch_vlans_ports() {
 				json_get_vars ports
 			json_select ..
 
+			vlan_num=$((role - 1))
 			uci -q batch <<-EOF
 				add network switch_vlan
 				set network.@switch_vlan[-1].device='$switch'
-				set network.@switch_vlan[-1].vlan='$role'
+				set network.@switch_vlan[-1].vlan='$vlan_num'
 				set network.@switch_vlan[-1].ports='$ports'
 			EOF
 		done
