--- a/target/linux/ipq40xx/base-files/etc/board.d/02_network
+++ b/target/linux/ipq40xx/base-files/etc/board.d/02_network
@@ -34,7 +34,9 @@ ipq40xx_setup_interfaces()
 	netgear,ex6100v2|\
 	netgear,ex6150v2|\
 	zyxel,wre6606)
-		ucidef_set_interface_lan "eth0"
+		ucidef_set_interfaces_lan_wan "eth0" "eth0.1"
+		ucidef_set_interface "dtdlink" device "eth0.2" protocol "static"
+		ucidef_set_interface "wifi" device "wlan0" protocol "static"
 		;;
 	aruba,ap-303h|\
 	teltonika,rutx10)
@@ -63,7 +65,8 @@ ipq40xx_setup_interfaces()
 	zyxel,nbg6617)
 		ucidef_set_interfaces_lan_wan "eth0" "eth1"
 		ucidef_add_switch "switch0" \
-			"0u@eth0" "1:lan:4" "2:lan:3" "3:lan:2" "4:lan:1"
+			"0@eth0"  "5:lan" "4:lan:2" "3:lan:3" "2:lan:4" "1t:dtdlink:5"
+		ucidef_set_interface "wifi" device "wlan0" protocol "static"
 		;;
 	avm,fritzbox-4040|\
 	linksys,ea6350v3|\
--- a/target/linux/ath79/nand/base-files/etc/board.d/02_network
+++ b/target/linux/ath79/nand/base-files/etc/board.d/02_network
@@ -26,7 +26,7 @@ ath79_setup_interfaces()
 	glinet,gl-ar750s-nor-nand)
 		ucidef_add_switch "switch0" \
 			"0@eth0" "1:wan" "3t:dtdlink" "2:lan"
-		ucidef_set_interface "wifi" device "wlan1" protocol "static"
+		ucidef_set_interface "wifi" device "wlan0" protocol "static"
 		;;
 	glinet,gl-xe300)
 		ucidef_set_interface_wan "eth1"
