--- a/target/linux/ath79/generic/base-files/etc/board.d/02_network
+++ b/target/linux/ath79/generic/base-files/etc/board.d/02_network
@@ -94,7 +94,9 @@ ath79_setup_interfaces()
 	ubnt,unifi|\
 	wd,mynet-wifi-rangeextender|\
 	winchannel,wb2000)
-		ucidef_set_interface_lan "eth0"
+		ucidef_set_interfaces_lan_wan "eth0" "eth0.1"
+		ucidef_set_interface "wifi" device "wlan0" protocol "static"
+		ucidef_set_interface "dtdlink" device "eth0.2" protocol "static"
 		;;
 	airtight,c-75)
 		ucidef_add_switch "switch0" \
@@ -153,8 +155,9 @@ ath79_setup_interfaces()
 	tplink,tl-wr842n-v3|\
 	ubnt,airrouter)
 		ucidef_set_interface_wan "eth1"
+		ucidef_set_interface "wifi" device "wlan0" "protocol" "static"
 		ucidef_add_switch "switch0" \
-			"0@eth0" "1:lan:4" "2:lan:3" "3:lan:2" "4:lan:1"
+			"0@eth0" "4:lan:1" "3:lan:2" "2:lan:3" "1t:unused" "1t:dtdlink"
 		;;
 	buffalo,wzr-hp-g300nh-rb|\
 	buffalo,wzr-hp-g300nh-s|\
@@ -185,7 +188,10 @@ ath79_setup_interfaces()
 	tplink,wbs510-v2|\
 	ubnt,nanostation-m|\
 	ubnt,routerstation)
-		ucidef_set_interfaces_lan_wan "eth1" "eth0"
+		ucidef_set_interface_lan "eth0 eth1"
+		ucidef_set_interface_wan "eth0.1 eth1.1"
+		ucidef_set_interface "dtdlink" device "eth0.2 eth1.2" protocol "static"
+		ucidef_set_interface "wifi" device "wlan0" protocol "static"
 		;;
 	comfast,cf-e560ac)
 		ucidef_set_interface_wan "eth1"
@@ -425,10 +431,6 @@ ath79_setup_interfaces()
 		ucidef_add_switch "switch0" \
 			"0@eth1" "2:lan:3" "3:lan:2" "4:lan:1"
 		;;
-	ubnt,nanostation-m-xw)
-		ucidef_add_switch "switch0" \
-			"0@eth0" "5:lan" "1:wan"
-		;;
 	ubnt,unifiac-mesh-pro|\
 	ubnt,unifiac-pro)
 		ucidef_add_switch "switch0" \
@@ -444,8 +446,15 @@ ath79_setup_interfaces()
 		ucidef_add_switch "switch0" \
 			"0@eth0" "3:lan:1" "4:lan:2"
 		;;
+        glinet,gl-ar150)
+                ucidef_set_interface "wifi" device "wlan0" protocol "static"
+                ucidef_set_interfaces_lan_wan "eth0" "eth1"
+                ucidef_set_interface "dtdlink" device "eth0.2" protocol "static"
+                ;;
 	*)
-		ucidef_set_interfaces_lan_wan "eth0" "eth1"
+		ucidef_set_interface "wifi" device "wlan0" protocol "static"
+		ucidef_set_interfaces_lan_wan "eth0" "eth0.1"
+		ucidef_set_interface "dtdlink" device "eth0.2" protocol "static"
 		;;
 	esac
 }
