--- a/target/linux/ipq40xx/base-files/etc/board.d/02_network
+++ b/target/linux/ipq40xx/base-files/etc/board.d/02_network
@@ -30,12 +30,16 @@
 	engenius,emd1|\
 	meraki,mr33|\
 	mikrotik,lhgg-60ad|\
-	mikrotik,sxtsq-5-ac|\
 	netgear,ex6100v2|\
 	netgear,ex6150v2|\
 	zyxel,wre6606)
 		ucidef_set_interface_lan "eth0"
 		;;
+	mikrotik,sxtsq-5-ac)
+		ucidef_set_interfaces_lan_wan "lan" "lan.1"
+		ucidef_set_interface "wifi" device "wlan0" protocol "static"
+		ucidef_set_interface "dtdlink" device "lan.2" protocol "static"
+		;;
 	aruba,ap-303h|\
 	teltonika,rutx10)
 		ucidef_set_interfaces_lan_wan "eth0" "eth1"

--- a/target/linux/ipq40xx/base-files/etc/board.d/02_network
+++ b/target/linux/ipq40xx/base-files/etc/board.d/02_network
@@ -61,9 +61,9 @@
 	p2w,r619ac-64m|\
 	p2w,r619ac-128m|\
 	zyxel,nbg6617)
-		ucidef_set_interfaces_lan_wan "eth0" "eth1"
-		ucidef_add_switch "switch0" \
-			"0u@eth0" "1:lan:4" "2:lan:3" "3:lan:2" "4:lan:1"
+		ucidef_set_interfaces_lan_wan "lan1 lan2 lan3" "wan"
+		ucidef_set_interface "dtdlink" device "lan4.2" protocol "static"
+		ucidef_set_interface "wifi" device "wlan0" protocol "static"
 		;;
 	avm,fritzbox-4040|\
 	linksys,ea6350v3|\
