--- a/package/utils/busybox/Config-defaults.in
+++ b/package/utils/busybox/Config-defaults.in
@@ -2450,7 +2450,7 @@ config BUSYBOX_DEFAULT_PING
 	default y
 config BUSYBOX_DEFAULT_PING6
 	bool
-	default y
+	default n
 config BUSYBOX_DEFAULT_FEATURE_FANCY_PING
 	bool
 	default y
@@ -2528,7 +2528,7 @@ config BUSYBOX_DEFAULT_TRACEROUTE
 	default y
 config BUSYBOX_DEFAULT_TRACEROUTE6
 	bool
-	default y
+	default n
 config BUSYBOX_DEFAULT_FEATURE_TRACEROUTE_VERBOSE
 	bool
 	default y
