Index: openwrt/package/utils/busybox/Config-defaults.in
===================================================================
--- openwrt.orig/package/utils/busybox/Config-defaults.in
+++ openwrt/package/utils/busybox/Config-defaults.in
@@ -2449,7 +2449,7 @@ config BUSYBOX_DEFAULT_PING
 	default y
 config BUSYBOX_DEFAULT_PING6
 	bool
-	default y
+	default n
 config BUSYBOX_DEFAULT_FEATURE_FANCY_PING
 	bool
 	default y
@@ -2527,7 +2527,7 @@ config BUSYBOX_DEFAULT_TRACEROUTE
 	default y
 config BUSYBOX_DEFAULT_TRACEROUTE6
 	bool
-	default y
+	default n
 config BUSYBOX_DEFAULT_FEATURE_TRACEROUTE_VERBOSE
 	bool
 	default y
