--- a/package/network/services/hostapd/files/wpad.init
+++ b/package/network/services/hostapd/files/wpad.init
@@ -24,7 +24,9 @@
 		procd_close_instance
 	fi
 
-	if [ -x "/usr/sbin/wpa_supplicant" ]; then
+	if [ "$(uci -q get wireless.@wifi-iface[0].encryption)" == "" -o "$(uci -q get wireless.@wifi-iface[0].encryption)" = "none" ] && [ "$(uci -q get wireless.@wifi-iface[1].encryption)" = "" -o "$(uci -q get wireless.@wifi-iface[1].encryption)" == "none" ]; then
+		true # Don't run supplicant to save some memory
+	elif [ -x "/usr/sbin/wpa_supplicant" ]; then
 		mkdir -p /var/run/wpa_supplicant
 		chown network:network /var/run/wpa_supplicant
 		procd_open_instance supplicant
