Index: openwrt/package/base-files/files/etc/init.d/sysctl
===================================================================
--- openwrt.orig/package/base-files/files/etc/init.d/sysctl
+++ openwrt/package/base-files/files/etc/init.d/sysctl
@@ -10,7 +10,7 @@ apply_defaults() {
 	if [ "$mem" -gt 65536 ]; then # 128M
 		min_free=16384
 	elif [ "$mem" -gt 32768 ]; then # 64M
-		min_free=8192
+		min_free=2048
 	else
 		min_free=1024
 		frag_low_thresh=393216
