--- a/package/firmware/ath10k-ct-firmware/Makefile
+++ b/package/firmware/ath10k-ct-firmware/Makefile
@@ -628,7 +628,7 @@
 $(eval $(call BuildPackage,ath10k-firmware-qca9984-ct))
 $(eval $(call BuildPackage,ath10k-firmware-qca9984-ct-full-htt))
 $(eval $(call BuildPackage,ath10k-firmware-qca9984-ct-htt))
-$(eval $(call BuildPackage,ath10k-firmware-qca4019-ct))
+#$(eval $(call BuildPackage,ath10k-firmware-qca4019-ct))
 $(eval $(call BuildPackage,ath10k-firmware-qca4019-ct-full-htt))
 $(eval $(call BuildPackage,ath10k-firmware-qca4019-ct-htt))
 $(eval $(call BuildPackage,ath10k-firmware-qca9888-ct))
