--- a/target/linux/ath79/image/generic-ubnt.mk
+++ b/target/linux/ath79/image/generic-ubnt.mk
@@ -157,6 +157,22 @@
 endef
 TARGET_DEVICES += ubnt_powerbeam-m5-xw
 
+define Device/ubnt_rocket-m-xw
+  $(Device/ubnt-xw)
+  DEVICE_MODEL := Rocket M5 XW
+  DEVICE_PACKAGES += rssileds
+  SUPPORTED_DEVICES += rocket-m-xw ubnt,rocket-m-xw loco-m-xw
+endef
+TARGET_DEVICES += ubnt_rocket-m-xw
+
+define Device/ubnt_rocket-m2-xw
+  $(Device/ubnt-xw)
+  DEVICE_MODEL := Rocket M2 XW
+  DEVICE_PACKAGES += rssileds
+  SUPPORTED_DEVICES += rocket-m-xw rocket-m2-xw loco-m-xw
+endef
+TARGET_DEVICES += ubnt_rocket-m2-xw
+
 define Device/ubnt_rocket-5ac-lite
   $(Device/ubnt-xc)
   SOC := qca9558

--- /dev/null
+++ b/target/linux/ath79/dts/ar9342_ubnt_rocket-m-xw.dts
@@ -0,0 +1,8 @@
+// SPDX-License-Identifier: GPL-2.0-only
+
+#include "ar9342_ubnt_bullet-m-xw.dts"
+
+/ {
+	compatible = "ubnt,rocket-m-xw", "ubnt,xw", "qca,ar9342";
+	model = "Ubiquiti Rocket M5 (XW)";
+};
--- /dev/null
+++ b/target/linux/ath79/dts/ar9342_ubnt_rocket-m2-xw.dts
@@ -0,0 +1,26 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+#include "ar9342_ubnt_xw.dtsi"
+
+/ {
+	compatible = "ubnt,rocket-m2-xw", "ubnt,xw", "qca,ar9342";
+	model = "Ubiquiti Rocket M2 (XW)";
+};
+
+&mdio0 {
+	status = "okay";
+
+	phy-mask = <0x1>;
+
+	phy1: ethernet-phy@1 {
+		reg = <1>;
+		phy-mode = "mii";
+		reset-gpios = <&gpio 0 GPIO_ACTIVE_LOW>;
+	};
+};
+
+&eth0 {
+	status = "okay";
+
+	phy-handle = <&phy1>;
+};
--- a/target/linux/ath79/image/generic-ubnt.mk
+++ b/target/linux/ath79/image/generic-ubnt.mk
@@ -28,9 +28,6 @@
 define Device/ubnt_bullet-m-xw
   $(Device/ubnt-xw)
   DEVICE_MODEL := Bullet M
-  DEVICE_ALT0_VENDOR := Ubiquiti
-  DEVICE_ALT0_MODEL := Rocket M
-  DEVICE_ALT0_VARIANT := XW
   DEVICE_PACKAGES += rssileds
   SUPPORTED_DEVICES += bullet-m-xw
 endef
