--- a/target/linux/ath79/tiny/config-default
+++ b/target/linux/ath79/tiny/config-default
@@ -1,5 +1,5 @@
 CONFIG_GRO_CELLS=y
-CONFIG_MTD_SPI_NOR_USE_4K_SECTORS=y
+CONFIG_MTD_SPI_NOR_USE_4K_SECTORS=n
 CONFIG_NET_DEVLINK=y
 CONFIG_NET_DSA=y
 CONFIG_NET_DSA_MV88E6060=y
--- a/target/linux/ath79/tiny/base-files/etc/board.d/02_network
+++ b/target/linux/ath79/tiny/base-files/etc/board.d/02_network
@@ -108,7 +108,8 @@
 	ubnt,airrouter)
 		ucidef_set_interface_wan "eth1"
 		ucidef_add_switch "switch0" \
-			"0@eth0" "1:lan:4" "2:lan:3" "3:lan:2" "4:lan:1"
+			"0@eth0" "1t:dummy" "1t:dtdlink" "4:lan" "3:lan" "2:lan"
+		ucidef_set_interface "wifi" device "wlan0" "protocol" "static"
 		;;
 	ubnt,nanostation-m)
 		ucidef_set_interface_lan "eth0 eth1"
--- /dev/null
+++ b/target/linux/ath79/tiny/base-files/etc/board.d/03_gpio_switches
@@ -0,0 +1,19 @@
+#
+# Copyright (C) 2018 OpenWrt.org
+#
+
+. /lib/functions/uci-defaults.sh
+
+board_config_update
+
+board=$(board_name)
+
+case "$board" in
+ubnt,nanostation-m)
+	ucidef_add_gpio_switch "poe_passthrough" "PoE Passthrough" "8"
+	;;
+esac
+
+board_config_flush
+
+exit 0
