--- a/target/linux/ath79/image/generic-ubnt.mk
+++ b/target/linux/ath79/image/generic-ubnt.mk
@@ -117,6 +117,22 @@
 endef
 TARGET_DEVICES += ubnt_nanostation-loco-m-xw
 
+define Device/ubnt_nanobeam-m5-19
+  $(Device/ubnt-xw)
+  DEVICE_MODEL := Nanobeam M5 19
+  DEVICE_PACKAGES += rssileds
+  SUPPORTED_DEVICES += loco-m-xw ubnt,nanostation-loco-m-xw
+endef
+TARGET_DEVICES += ubnt_nanobeam-m5-19
+
+define Device/ubnt_nanobeam-m5-16
+  $(Device/ubnt-xw)
+  DEVICE_MODEL := Nanobeam M5 16
+  DEVICE_PACKAGES += rssileds
+  SUPPORTED_DEVICES += loco-m-xw ubnt,nanostation-loco-m-xw
+endef
+TARGET_DEVICES += ubnt_nanobeam-m5-16
+
 define Device/ubnt_nanostation-m-xw
   $(Device/ubnt-xw)
   DEVICE_MODEL := Nanostation M
--- /dev/null
+++ b/target/linux/ath79/dts/ar9342_ubnt_nanobeam-m5-19.dts
@@ -0,0 +1,26 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+#include "ar9342_ubnt_xw.dtsi"
+
+/ {
+	compatible = "ubnt,nanobeam-m5-19", "ubnt,xw", "qca,ar9342";
+	model = "Ubiquiti Nanobeam M5 19";
+};
+
+&mdio0 {
+	status = "okay";
+
+	phy-mask = <0x1>;
+
+	phy1: ethernet-phy@1 {
+		reg = <1>;
+		phy-mode = "mii";
+		reset-gpios = <&gpio 0 GPIO_ACTIVE_LOW>;
+	};
+};
+
+&eth0 {
+	status = "okay";
+
+	phy-handle = <&phy1>;
+};
--- /dev/null
+++ b/target/linux/ath79/dts/ar9342_ubnt_nanobeam-m5-16.dts
@@ -0,0 +1,26 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+#include "ar9342_ubnt_xw.dtsi"
+
+/ {
+	compatible = "ubnt,nanobeam-m5-16", "ubnt,xw", "qca,ar9342";
+	model = "Ubiquiti Nanobeam M5 16";
+};
+
+&mdio0 {
+	status = "okay";
+
+	phy-mask = <0x1>;
+
+	phy1: ethernet-phy@1 {
+		reg = <1>;
+		phy-mode = "mii";
+		reset-gpios = <&gpio 0 GPIO_ACTIVE_LOW>;
+	};
+};
+
+&eth0 {
+	status = "okay";
+
+	phy-handle = <&phy1>;
+};
--- a/target/linux/ath79/generic/base-files/etc/board.d/02_network
+++ b/target/linux/ath79/generic/base-files/etc/board.d/02_network
@@ -96,6 +96,7 @@
 	ubnt,litebeam-ac-gen2|\
 	ubnt,nanobeam-ac|\
 	ubnt,nanobeam-ac-xc|\
+	ubnt,nanobeam-m5-1[69]|\
 	ubnt,nanostation-ac-loco|\
 	ubnt,nanostation-loco-m-xw|\
 	ubnt,powerbeam-5ac-500|\
