--- a/target/linux/ath79/generic/base-files/etc/board.d/02_network
+++ b/target/linux/ath79/generic/base-files/etc/board.d/02_network
@@ -539,6 +539,7 @@
                 ucidef_set_interface "dtdlink" device "eth0.2" protocol "static"
                 ucidef_set_interface "wifi" device "wlan0" protocol "static"
                 ;;
+	ubnt,powerbridge-m|\
 	ubnt,nanostation-m)
 		ucidef_set_interfaces_lan_wan "eth0 eth1" "eth0.1 eth1.1"
 		ucidef_set_interface "dtdlink" device "eth0.2 eth1.2" protocol "static"
@@ -546,7 +547,6 @@
 		;;
 	ubnt,bullet-m-ar7241|\
 	ubnt,nanobridge-m|\
-	ubnt,powerbridge-m|\
 	ubnt,rocket-m)
 		ucidef_set_interfaces_lan_wan "eth0" "eth0.1"
 		ucidef_set_interface "dtdlink" device "eth0.2" protocol "static"
