--- a/target/linux/ath79/image/generic-ubnt.mk
+++ b/target/linux/ath79/image/generic-ubnt.mk
@@ -79,6 +79,13 @@
 endef
 TARGET_DEVICES += ubnt_nanobeam-ac-gen2
 
+define Device/ubnt_nanobeam-2ac-13
+  $(Device/ubnt-2wa)
+  DEVICE_MODEL := NanoBeam 2AC 13
+  DEVICE_PACKAGES += kmod-ath10k-ct-smallbuffers ath10k-firmware-qca988x-ddwrt rssileds
+endef
+TARGET_DEVICES += ubnt_nanobeam-2ac-13
+
 define Device/ubnt_nanobeam-ac-xc
   $(Device/ubnt-xc)
   SOC := qca9558
--- /dev/null
+++ b/target/linux/ath79/dts/ar9342_ubnt_nanobeam-2ac-13.dts
@@ -0,0 +1,38 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "ar9342_ubnt_wa_1port.dtsi"
+
+/ {
+	compatible = "ubnt,nanobeam-2ac-13", "ubnt,wa", "qca,ar9342";
+	model = "Ubiquiti NanoBeam 2AC 13 (2WA)";
+
+	aliases {
+		led-boot = &led_rssi3;
+		led-failsafe = &led_rssi3;
+		led-upgrade = &led_rssi3;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		rssi0 {
+			label = "blue:rssi0";
+			gpios = <&gpio 11 GPIO_ACTIVE_LOW>;
+		};
+
+		rssi1 {
+			label = "blue:rssi1";
+			gpios = <&gpio 16 GPIO_ACTIVE_LOW>;
+		};
+
+		rssi2 {
+			label = "blue:rssi2";
+			gpios = <&gpio 13 GPIO_ACTIVE_LOW>;
+		};
+
+		led_rssi3: rssi3 {
+			label = "blue:rssi3";
+			gpios = <&gpio 14 GPIO_ACTIVE_LOW>;
+		};
+	};
+};
--- a/target/linux/ath79/generic/base-files/etc/board.d/02_network
+++ b/target/linux/ath79/generic/base-files/etc/board.d/02_network
@@ -103,6 +103,7 @@
 	ubnt,lap-120|\
 	ubnt,litebeam-ac-gen2|\
 	ubnt,nanobeam-ac|\
+	ubnt,nanobeam-2ac-13|\
 	ubnt,nanobeam-ac-xc|\
 	ubnt,nanobeam-ac-gen2-xc|\
 	ubnt,nanobeam-m5-1[69]|\
