--- a/target/linux/ath79/generic/base-files/etc/board.d/02_network
+++ b/target/linux/ath79/generic/base-files/etc/board.d/02_network
@@ -115,7 +115,6 @@ ath79_setup_interfaces()
 	ubnt,nanobeam-ac-xc|\
 	ubnt,nanobeam-ac-gen2-xc|\
 	ubnt,nanobeam-m5-1[69]|\
-	ubnt,nanostation-ac-loco|\
 	ubnt,nanostation-loco-m-xw|\
 	ubnt,powerbeam-5ac-500|\
 	ubnt,powerbeam-5ac-400|\
@@ -614,7 +613,8 @@ ath79_setup_interfaces()
 		;;
 	ubnt,nanostation-m-xw|\
 	ubnt,nanobeam-ac-gen2|\
-	ubnt,nanostation-ac)
+	ubnt,nanostation-ac|\
+	ubnt,nanostation-ac-loco)
 		ucidef_set_interfaces_lan_wan "eth0" "eth0.1"
 		ucidef_set_interface "dtdlink" device "eth0.2" protocol "static"
 		ucidef_set_interface "wifi" device "wlan0" protocol "static"
