--- a/target/linux/ramips/mt76x8/base-files/etc/board.d/02_network
+++ b/target/linux/ramips/mt76x8/base-files/etc/board.d/02_network
@@ -100,8 +100,11 @@
 		;;
 	cudy,m1200-v1|\
 	cudy,tr1200-v1)
-		ucidef_add_switch "switch0" \
-			"0:lan" "1:wan" "6@eth0"
+		ucidef_add_switch "switch0"
+		ucidef_add_switch_attr "switch0" "enable" "false"
+		ucidef_set_interfaces_lan_wan "eth0" "eth0.1"
+		ucidef_set_interface "dtdlink" device "eth0.2" protocol "static"
+		ucidef_set_interface "wifi" device "wlan0" protocol "static"
 		;;
 	cudy,wr1000)
 		ucidef_add_switch "switch0" \

--- a/target/linux/mediatek/filogic/base-files/etc/board.d/02_network
+++ b/target/linux/mediatek/filogic/base-files/etc/board.d/02_network
@@ -51,7 +51,6 @@
 	creatlentem,clt-r30b1|\
 	creatlentem,clt-r30b1-112m|\
 	creatlentem,clt-r30b1-ubi|\
-	cudy,wr3000-v1|\
 	huasifei,wh3000r-nand|\
 	jcg,q30-pro|\
 	keenetic,kn-3711|\
@@ -63,6 +62,19 @@
 	tenbay,wr3000k)
 		ucidef_set_interfaces_lan_wan "lan1 lan2 lan3" wan
 		;;
+	cudy,wr3000-v1)
+		ucidef_set_interfaces_lan_wan "lan1 lan2" "wan"
+		ucidef_set_interface "dtdlink" device "lan3.2" protocol "static"
+		ucidef_set_interface "wifi" device "wlan0" protocol "static"
+		;;
+	cudy,wr3000e-v1|\
+	cudy,wr3000h-v1|\
+	cudy,wr3000p-v1|\
+	cudy,wr3000s-v1)
+		ucidef_set_interfaces_lan_wan "lan1 lan2 lan3" "wan"
+		ucidef_set_interface "dtdlink" device "lan4.2" protocol "static"
+		ucidef_set_interface "wifi" device "wlan0" protocol "static"
+		;;
 	asus,tuf-ax4200|\
 	iptime,ax3000sm|\
 	iptime,ax7800m-6e|\
