--- a/package/feeds/packages/libx264/Makefile
+++ b/package/feeds/packages/libx264/Makefile
@@ -14,7 +14,7 @@
 PKG_SOURCE_URL:=https://code.videolan.org/videolan/x264.git
 PKG_SOURCE_DATE:=2024-05-13
 PKG_SOURCE_VERSION:=4613ac3c15fd75cebc4b9f65b7fb95e70a3acce1
-PKG_MIRROR_HASH:=03d8ca3495185504a601e36bff017e1044cb3f6c32cb567fb12d6f9707bb78cd
+PKG_MIRROR_HASH:=ac55224b0b4eb459d3b594b6014aa97a48c8a3d603fc68959debef833fbeaf58
 
 PKG_MAINTAINER:=Adrian Panella <ianchi74@outlook.com>
 PKG_LICENSE:=GPL-2.0-or-later
