--- a/target/linux/ath79/image/mikrotik.mk
+++ b/target/linux/ath79/image/mikrotik.mk
@@ -92,6 +92,7 @@
   SOC := qca9533
   DEVICE_MODEL := RouterBOARD LHG 2nD (LHG 2)
   IMAGE_SIZE := 16256k
+  SUPPORTED_DEVICES += rb-lhg-2nd
 endef
 TARGET_DEVICES += mikrotik_routerboard-lhg-2nd
 
@@ -101,6 +102,7 @@
   DEVICE_MODEL := RouterBOARD LHG 5nD (LHG 5)
   DEVICE_PACKAGES += rssileds
   IMAGE_SIZE := 16256k
+  SUPPORTED_DEVICES += rb-lhg-5nd rb-lhg-5hpnd
 endef
 TARGET_DEVICES += mikrotik_routerboard-lhg-5nd
 
@@ -127,6 +129,7 @@
   DEVICE_MODEL := RouterBOARD SXT 5nD (SXT Lite5)
   DEVICE_PACKAGES += rssileds
   IMAGE_SIZE := 16256k
+  SUPPORTED_DEVICES += rb-lhg-5nd rb-lhg-5hpnd
 endef
 TARGET_DEVICES += mikrotik_routerboard-sxt-5nd
 