--- a/target/linux/ath79/image/mikrotik.mk
+++ b/target/linux/ath79/image/mikrotik.mk
@@ -92,6 +92,7 @@
   SOC := qca9533
   DEVICE_MODEL := RouterBOARD LHG 2nD (LHG 2)
   IMAGE_SIZE := 16256k
+  SUPPORTED_DEVICES += rb-lhg-2nd rb-lhg-2hpnd rb-lhg-2hpnd-xl rb-lhg-5hpnd rb-lhg-5hpnd-xl
 endef
 TARGET_DEVICES += mikrotik_routerboard-lhg-2nd
 
@@ -101,6 +102,7 @@
   DEVICE_MODEL := RouterBOARD LHG 5nD (LHG 5)
   DEVICE_PACKAGES += rssileds
   IMAGE_SIZE := 16256k
+  SUPPORTED_DEVICES += rb-lhg-5nd rb-lhg-5hpnd rb-lhg-5hpnd-xl
 endef
 TARGET_DEVICES += mikrotik_routerboard-lhg-5nd
 
@@ -127,6 +129,7 @@
   DEVICE_MODEL := RouterBOARD SXT 5nD (SXT Lite5)
   DEVICE_PACKAGES += rssileds
   IMAGE_SIZE := 16256k
+  SUPPORTED_DEVICES += rb-lhg-5nd rb-lhg-5hpnd
 endef
 TARGET_DEVICES += mikrotik_routerboard-sxt-5nd

--- a/target/linux/ath79/image/generic-ubnt.mk
+++ b/target/linux/ath79/image/generic-ubnt.mk
@@ -153,7 +153,7 @@
   $(Device/ubnt-xw)
   DEVICE_MODEL := PowerBeam M5
   DEVICE_PACKAGES += rssileds
-  SUPPORTED_DEVICES += loco-m-xw
+  SUPPORTED_DEVICES += loco-m-xw rocket-m-xw
 endef
 TARGET_DEVICES += ubnt_powerbeam-m5-xw
