--- a/target/linux/ath79/image/mikrotik.mk
+++ b/target/linux/ath79/image/mikrotik.mk
@@ -96,6 +96,33 @@
 endef
 TARGET_DEVICES += mikrotik_routerboard-lhg-2nd
 
+define Device/mikrotik_routerboard-lhg-2nd-xl
+  $(Device/mikrotik_nor)
+  SOC := qca9533
+  DEVICE_MODEL := RouterBOARD LHG 2nD-XL (LHG 2)
+  IMAGE_SIZE := 16256k
+  SUPPORTED_DEVICES += rb-lhg-2nd-xl mikrotik,routerboard-lhg-2nd
+endef
+TARGET_DEVICES += mikrotik_routerboard-lhg-2nd-xl
+
+define Device/mikrotik_routerboard-sxt-2nd
+  $(Device/mikrotik_nor)
+  SOC := qca9533
+  DEVICE_MODEL := RouterBOARD SXT 2nD (SXT Lite2)
+  IMAGE_SIZE := 16256k
+  SUPPORTED_DEVICES += rb-sxt-2nd rb-lhg-5nd
+endef
+TARGET_DEVICES += mikrotik_routerboard-sxt-2nd
+
+define Device/mikrotik_routerboard-ldf-2nd
+  $(Device/mikrotik_nor)
+  SOC := qca9533
+  DEVICE_MODEL := RouterBOARD LDF 2nD
+  IMAGE_SIZE := 16256k
+  SUPPORTED_DEVICES += rb-ldf-2nd rb-sxt-2nd rb-lhg-2nd
+endef
+TARGET_DEVICES += mikrotik_routerboard-ldf-2nd
+
 define Device/mikrotik_routerboard-lhg-5nd
   $(Device/mikrotik_nor)
   SOC := ar9344
@@ -106,6 +124,26 @@
 endef
 TARGET_DEVICES += mikrotik_routerboard-lhg-5nd
 
+define Device/mikrotik_routerboard-lhg-5hpnd
+  $(Device/mikrotik_nor)
+  SOC := ar9344
+  DEVICE_MODEL := RouterBOARD LHG 5HPnD (LHG 5)
+  DEVICE_PACKAGES += rssileds
+  IMAGE_SIZE := 16256k
+  SUPPORTED_DEVICES += rb-lhg-5hpnd mikrotik,routerboard-lhg-5nd
+endef
+TARGET_DEVICES += mikrotik_routerboard-lhg-5hpnd
+
+define Device/mikrotik_routerboard-lhg-5hpnd-xl
+  $(Device/mikrotik_nor)
+  SOC := ar9344
+  DEVICE_MODEL := RouterBOARD LHG 5HPnD-XL (LHG 5)
+  DEVICE_PACKAGES += rssileds
+  IMAGE_SIZE := 16256k
+  SUPPORTED_DEVICES += rb-lhg-5hpnd-xl mikrotik,routerboard-lhg-5nd
+endef
+TARGET_DEVICES += mikrotik_routerboard-lhg-5hpnd-xl
+
 define Device/mikrotik_routerboard-map-2nd
   $(Device/mikrotik_nor)
   SOC := qca9533
--- /dev/null
+++ b/target/linux/ath79/dts/qca9533_mikrotik_routerboard-lhg-2nd-xl.dts
@@ -0,0 +1,8 @@
+// SPDX-License-Identifier: GPL-2.0-only
+
+#include "qca9533_mikrotik_routerboard-lhg-2nd.dts"
+
+/ {
+        compatible = "mikrotik,routerboard-lhg-2nd-xl", "qca,qca9533";
+        model = "MikroTik RouterBOARD LHG 2nD-XL";
+};
--- /dev/null
+++ b/target/linux/ath79/dts/ar9344_mikrotik_routerboard-lhg-5hpnd.dts
@@ -0,0 +1,8 @@
+// SPDX-License-Identifier: GPL-2.0-only
+
+#include "ar9344_mikrotik_routerboard-lhg-5nd.dts"
+
+/ {
+        compatible = "mikrotik,routerboard-lhg-5hpnd", "qca,qca9533";
+        model = "MikroTik RouterBOARD LHG 5HPnD";
+};
--- /dev/null
+++ b/target/linux/ath79/dts/ar9344_mikrotik_routerboard-lhg-5hpnd-xl.dts
@@ -0,0 +1,8 @@
+// SPDX-License-Identifier: GPL-2.0-only
+
+#include "ar9344_mikrotik_routerboard-lhg-5nd.dts"
+
+/ {
+        compatible = "mikrotik,routerboard-lhg-5hpnd-xl", "qca,qca9533";
+        model = "MikroTik RouterBOARD LHG 5HPnD-XL";
+};
--- /dev/null
+++ b/target/linux/ath79/dts/qca9533_mikrotik_routerboard-sxt-2nd.dts
@@ -0,0 +1,8 @@
+// SPDX-License-Identifier: GPL-2.0-only
+
+#include "qca9533_mikrotik_routerboard-lhg-2nd.dts"
+
+/ {
+        compatible = "mikrotik,routerboard-sxt-2nd", "qca,qca9533";
+        model = "MikroTik RouterBOARD SXT 2nD (SXT Lite2)";
+};
--- /dev/null
+++ b/target/linux/ath79/dts/qca9533_mikrotik_routerboard-ldf-2nd.dts
@@ -0,0 +1,8 @@
+// SPDX-License-Identifier: GPL-2.0-only
+
+#include "qca9533_mikrotik_routerboard-lhg-2nd.dts"
+
+/ {
+        compatible = "mikrotik,routerboard-ldf-2nd", "qca,qca9533";
+        model = "MikroTik RouterBOARD LDF 2nD";
+};
--- a/target/linux/ath79/mikrotik/base-files/etc/board.d/02_network
+++ b/target/linux/ath79/mikrotik/base-files/etc/board.d/02_network
@@ -17,7 +17,12 @@
 	mikrotik,routerboard-912uag-2hpnd|\
 	mikrotik,routerboard-912uag-5hpnd|\
 	mikrotik,routerboard-lhg-2nd|\
+	mikrotik,routerboard-lhg-2nd-xl|\
 	mikrotik,routerboard-lhg-5nd|\
+	mikrotik,routerboard-lhg-5hpnd|\
+	mikrotik,routerboard-lhg-5hpnd-xl|\
+	mikrotik,routerboard-sxt-2nd|\
+	mikrotik,routerboard-ldf-2nd|\
 	mikrotik,routerboard-mapl-2nd|\
 	mikrotik,routerboard-sxt-5nd-r2|\
 	mikrotik,routerboard-sxt-5nd|\
@@ -73,7 +77,12 @@
 	mikrotik,routerboard-912uag-2hpnd|\
 	mikrotik,routerboard-912uag-5hpnd|\
 	mikrotik,routerboard-lhg-2nd|\
+	mikrotik,routerboard-lhg-2nd-xl|\
 	mikrotik,routerboard-lhg-5nd|\
+	mikrotik,routerboard-lhg-5hpnd|\
+	mikrotik,routerboard-lhg-5hpnd-xl|\
+	mikrotik,routerboard-sxt-2nd|\
+	mikrotik,routerboard-ldf-2nd|\
 	mikrotik,routerboard-map-2nd|\
 	mikrotik,routerboard-mapl-2nd|\
 	mikrotik,routerboard-sxt-5nd-r2|\
--- a/target/linux/ath79/mikrotik/base-files/etc/hotplug.d/firmware/10-ath9k-eeprom
+++ b/target/linux/ath79/mikrotik/base-files/etc/hotplug.d/firmware/10-ath9k-eeprom
@@ -30,9 +30,14 @@
 	mikrotik,routerboard-912uag-2hpnd|\
 	mikrotik,routerboard-912uag-5hpnd|\
 	mikrotik,routerboard-lhg-2nd|\
+	mikrotik,routerboard-lhg-2nd-xl|\
 	mikrotik,routerboard-lhg-5nd|\
+	mikrotik,routerboard-lhg-5hpnd|\
+	mikrotik,routerboard-lhg-5hpnd-xl|\
+	mikrotik,routerboard-sxt-2nd|\
 	mikrotik,routerboard-sxt-5nd|\
 	mikrotik,routerboard-sxt-5hpnd|\
+	mikrotik,routerboard-ldf-2nd|\
 	mikrotik,routerboard-ldf-5nd|\
 	mikrotik,routerboard-sxt-5nd-r2|\
 	mikrotik,routerboard-wap-2nd|\
