--- a/target/linux/ipq40xx/base-files/etc/board.d/03_gpio_switches
+++ b/target/linux/ipq40xx/base-files/etc/board.d/03_gpio_switches
@@ -7,7 +7,7 @@
 
 case "$board" in
 aruba,ap-303h)
-	ucidef_add_gpio_switch "poe_passtrough" "POE passtrough disable" "546" "1"
+	ucidef_add_gpio_switch "poe_passthrough" "POE passthrough disable" "546" "1"
 	;;
 cellc,rtl30vw)
 	ucidef_add_gpio_switch "w_disable" "W_DISABLE mPCIE pin" "398" "1"
@@ -16,16 +16,16 @@
 	ucidef_add_gpio_switch "usb_vcc" "USB power" "401" "0"
 	;;
 cilab,meshpoint-one)
-	ucidef_add_gpio_switch "poe_passtrough" "POE passtrough enable" "513" "1"
+	ucidef_add_gpio_switch "poe_passthrough" "POE passthrough enable" "513" "1"
 	;;
 compex,wpj428)
 	ucidef_add_gpio_switch "sim_card_select" "SIM card select" "3" "0"
 	;;
 mikrotik,cap-ac)
-	ucidef_add_gpio_switch "poe_passtrough" "POE passtrough enable" "514" "0"
+	ucidef_add_gpio_switch "poe_passthrough" "POE passthrough enable" "514" "0"
 	;;
 mikrotik,hap-ac3)
-	ucidef_add_gpio_switch "poe_passtrough" "PoE Passthrough" "552" "0"
+	ucidef_add_gpio_switch "poe_passthrough" "PoE Passthrough" "552" "0"
 	;;
 mikrotik,hap-ac3-lte6-kit)
 	ucidef_add_gpio_switch "lte_ant_sw1" "LTE Antenna SW 1" "557" "0"
