--- a/target/linux/ath79/generic/base-files/etc/board.d/02_network
+++ b/target/linux/ath79/generic/base-files/etc/board.d/02_network
@@ -538,6 +538,7 @@
 			"0@eth0" "3:lan:1" "4:lan:2"
 		;;
         glinet,gl-ar150)
+        glinet,gl-ar300m16)
                 ucidef_set_interfaces_lan_wan "eth0" "eth1"
                 ucidef_set_interface "dtdlink" device "eth0.2" protocol "static"
                 ucidef_set_interface "wifi" device "wlan0" protocol "static"
