--- a/target/linux/ipq40xx/base-files/etc/board.d/03_gpio_switches
+++ b/target/linux/ipq40xx/base-files/etc/board.d/03_gpio_switches
@@ -13,16 +13,16 @@
 	ucidef_add_gpio_switch "usb_vcc" "USB power" "401" "0"
 	;;
 cilab,meshpoint-one)
-	ucidef_add_gpio_switch "poe_passtrough" "POE passtrough enable" "413" "1"
+	ucidef_add_gpio_switch "poe_passthrough" "POE passthrough enable" "413" "1"
 	;;
 compex,wpj428)
 	ucidef_add_gpio_switch "sim_card_select" "SIM card select" "3" "0"
 	;;
 mikrotik,cap-ac)
-	ucidef_add_gpio_switch "poe_passtrough" "POE passtrough enable" "414" "0"
+	ucidef_add_gpio_switch "poe_passthrough" "POE passthrough enable" "414" "0"
 	;;
 mikrotik,hap-ac3)
-	ucidef_add_gpio_switch "poe_passtrough" "PoE Passthrough" "452" "0"
+	ucidef_add_gpio_switch "poe_passthrough" "PoE Passthrough" "452" "0"
 	;;
 mikrotik,hap-ac3-lte6-kit)
