--- a/target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-fritzrepeater-1200.dts
+++ b/target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-fritzrepeater-1200.dts
@@ -260,13 +288,3 @@
 &psgmiiphy {
 	status = "disabled";
 };
-
-&wifi0 {
-	status = "okay";
-	qcom,ath10k-calibration-variant = "AVM-FRITZRepeater-1200";
-};
-
-&wifi1 {
-	status = "okay";
-	qcom,ath10k-calibration-variant = "AVM-FRITZRepeater-1200";
-};
