--- a/target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4018-dap-2610.dts
+++ b/target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4018-dap-2610.dts
@@ -51,16 +46,6 @@
 		watchdog@b017000 {
 			status = "okay";
 		};
-
-		ess-switch@c000000 {
-			status = "okay";
-			switch_lan_bmp = <0x20>;
-			switch_wan_bmp = <0x00>;
-		};
-
-		ess-psgmii@98000 {
-			status = "okay";
-		};
 	};
 
 	leds {
