--- a/target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4028-wpj428.dts
+++ b/target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4028-wpj428.dts
@@ -227,6 +227,14 @@
 				precal_art_5000: precal@5000 {
 					reg = <0x5000 0x2f20>;
 				};
+
+				macaddr_art_e010: mac-address@e010 {
+					reg = <0xe010 0x6>;
+				};
+
+				macaddr_art_e018: mac-address@e018 {
+					reg = <0xe018 0x6>;
+				};
 			};
 			partition6@180000 {
 				compatible = "denx,fit";
