--- a/target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4018-ap120c-ac.dts
+++ b/target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4018-ap120c-ac.dts
@@ -277,22 +262,6 @@
 	#gpio-cells = <2>;
 };
 
-&gmac0 {
-	qcom,forced_duplex = <1>;
-	qcom,forced_speed = <1000>;
-	qcom,phy_mdio_addr = <3>;
-	qcom,poll_required = <1>;
-	vlan_tag = <1 0x10>;
-};
-
-&gmac1 {
-	qcom,forced_duplex = <1>;
-	qcom,forced_speed = <1000>;
-	qcom,phy_mdio_addr = <4>;
-	qcom,poll_required = <1>;
-	vlan_tag = <2 0x20>;
-};
-
 &tlmm {
 	i2c0_pins: i2c0_pinmux {
 		mux_i2c {
diff --git a/target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4018-cap-ac.dts b/target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4018-cap-ac.dts
index b18ba4d46a279..515d71b456e94 100644
