--- a/target/linux/ipq40xx/base-files/etc/board.d/02_network
+++ b/target/linux/ipq40xx/base-files/etc/board.d/02_network
@@ -12,27 +12,57 @@ ipq40xx_setup_interfaces()
 
 	case "$board" in
 	8dev,habanero-dvk|\
+	avm,fritzbox-4040|\
+	asus,rt-ac58u|\
+	linksys,ea6350v3|\
+	linksys,ea8300|\
+	linksys,mr8300|\
+	mikrotik,hap-ac2|\
+	mikrotik,hap-ac3|\
+	p2w,r619ac-64m|\
+	p2w,r619ac-128m|\
+	zyxel,nbg6617)
+		ucidef_set_interfaces_lan_wan "lan1 lan2 lan3 lan4" "wan"
+		;;
 	8dev,jalapeno|\
 	alfa-network,ap120c-ac|\
+	asus,map-ac2200|\
+	cilab,meshpoint-one|\
+	edgecore,ecw5211|\
+	glinet,gl-b2200|\
+	mikrotik,cap-ac|\
+	netgear,wac510)
+		ucidef_set_interfaces_lan_wan "lan" "wan"
+		;;
+	aruba,ap-303|\
+	avm,fritzrepeater-1200|\
+	meraki,mr33|\
+    meraki,mr74|\
+	mikrotik,lhgg-60ad|\
+	mikrotik,sxtsq-5-ac)
+		ucidef_set_interface_lan "lan"
+		;;
+	avm,fritzbox-7530)
+		ucidef_set_interface_lan "lan1 lan2 lan3 lan4"
+		;;
+	cellc,rtl30vw)
+		ucidef_set_interface_lan "lan1 lan2"
+		;;
+	glinet,gl-b1300)
+		ucidef_set_interfaces_lan_wan "lan1 lan2" "wan"
+		;;
 	engenius,emr3500|\
 	engenius,ens620ext|\
 	luma,wrtq-329acn|\
-	netgear,wac510|\
 	plasmacloud,pa1200|\
 	plasmacloud,pa2200)
 		ucidef_set_interfaces_lan_wan "eth0" "eth1"
 		;;
-	aruba,ap-303|\
 	aruba,ap-365|\
-	avm,fritzrepeater-1200|\
 	dlink,dap-2610 |\
 	engenius,eap1300|\
 	engenius,emd1|\
 	extreme-networks,ws-ap3915i|\
-	meraki,mr33|\
-	meraki,mr74|\
-	mikrotik,lhgg-60ad|\
-	mikrotik,sxtsq-5-ac|\
 	netgear,ex6100v2|\
 	netgear,ex6150v2|\
 	zyxel,wre6606)
