--- a/target/linux/ipq40xx/files/drivers/net/dsa/qca/qca8k-ipq4019.h
+++ b/target/linux/ipq40xx/files/drivers/net/dsa/qca/qca8k-ipq4019.h
@@ -206,6 +207,28 @@
 #define PSGMIIPHY_VCO_CALIBRATION_CONTROL_REGISTER_2	0xa0
 #define   PSGMIIPHY_REG_PLL_VCO_CALIB_READY		BIT(0)
 
+#define   QCA8K_PSGMII_CALB_NUM				100
+#define   MII_QCA8075_SSTATUS				0x11
+#define   QCA8075_PHY_SPEC_STATUS_LINK			BIT(10)
+#define   QCA8075_MMD7_CRC_AND_PKTS_COUNT		0x8029
+#define   QCA8075_MMD7_PKT_GEN_PKT_NUMB			0x8021
+#define   QCA8075_MMD7_PKT_GEN_PKT_SIZE			0x8062
+#define   QCA8075_MMD7_PKT_GEN_CTRL			0x8020
+#define   QCA8075_MMD7_CNT_SELFCLR			BIT(1)
+#define   QCA8075_MMD7_CNT_FRAME_CHK_EN			BIT(0)
+#define   QCA8075_MMD7_PKT_GEN_START			BIT(13)
+#define   QCA8075_MMD7_PKT_GEN_INPROGR			BIT(15)
+#define   QCA8075_MMD7_IG_FRAME_RECV_CNT_HI		0x802a
+#define   QCA8075_MMD7_IG_FRAME_RECV_CNT_LO		0x802b
+#define   QCA8075_MMD7_IG_FRAME_ERR_CNT			0x802c
+#define   QCA8075_MMD7_EG_FRAME_RECV_CNT_HI		0x802d
+#define   QCA8075_MMD7_EG_FRAME_RECV_CNT_LO		0x802e
+#define   QCA8075_MMD7_EG_FRAME_ERR_CNT			0x802f
+#define   QCA8075_MMD7_MDIO_BRDCST_WRITE		0x8028
+#define   QCA8075_MMD7_MDIO_BRDCST_WRITE_EN 		BIT(15)
+#define   QCA8075_MDIO_BRDCST_PHY_ADDR			0x1f
+#define   QCA8075_PKT_GEN_PKTS_COUNT			4096
+
 enum {
 	QCA8K_PORT_SPEED_10M = 0,
 	QCA8K_PORT_SPEED_100M = 1,

From 553eb8c0823abd92ca8a4112599b513ae24eb388 Mon Sep 17 00:00:00 2001
From: Robert Marko <robert.marko@sartura.hr>
Date: Tue, 14 Dec 2021 01:00:00 +0100
Subject: [PATCH 16/19] ipq40xx: convert some boards to DSA

Convert IPQ40xx boards to DSA setup.

Signed-off-by: Leon M. George <leon@georgemail.eu>
Signed-off-by: Lech Perczak <lech.perczak@gmail.com>
Signed-off-by: Nick Hainke <vincent@systemli.org>
Signed-off-by: ChunAm See <z1250747241@gmail.com>
Signed-off-by: Jeff Kletsky <git-commits@allycomm.com>
Signed-off-by: Andrew Sim <andrewsimz@gmail.com>
Signed-off-by: Robert Marko <robert.marko@sartura.hr>
---
 .../ipq40xx/base-files/etc/board.d/01_leds    | 18 ++--
 .../ipq40xx/base-files/etc/board.d/02_network | 94 +++++++++----------
 .../base-files/etc/board.d/05_compat-version  | 18 ++++
 .../lib/preinit/05_set_iface_mac_ipq40xx.sh   |  4 +-
 .../preinit/06_set_preinit_iface_ipq40xx.sh   | 24 -----
 .../arm/boot/dts/qcom-ipq4018-ap120c-ac.dts   | 19 ++++
 .../arch/arm/boot/dts/qcom-ipq4018-cap-ac.dts | 18 ++++
 .../arm/boot/dts/qcom-ipq4018-ea6350v3.dts    | 28 ++++++
 .../arm/boot/dts/qcom-ipq4018-ecw5211.dts     | 20 ++++
 .../boot/dts/qcom-ipq4018-fritzbox-4040.dts   | 28 ++++++
 .../arm/boot/dts/qcom-ipq4018-hap-ac2.dts     | 32 +++++++
 .../arm/boot/dts/qcom-ipq4018-jalapeno.dtsi   | 22 +++++
 .../arm/boot/dts/qcom-ipq4018-nbg6617.dts     | 36 +++++++
 .../arm/boot/dts/qcom-ipq4018-rt-ac58u.dts    | 28 ++++++
 .../arm/boot/dts/qcom-ipq4018-sxtsq-5-ac.dts  | 17 ++++
 .../arch/arm/boot/dts/qcom-ipq4018-wac510.dts | 19 ++++
 .../boot/dts/qcom-ipq4019-fritzbox-7530.dts   | 24 +++++
 .../dts/qcom-ipq4019-fritzrepeater-1200.dts   | 42 ++++++---
 .../arm/boot/dts/qcom-ipq4019-gl-b2200.dts    | 24 +++++
 .../boot/dts/qcom-ipq4019-habanero-dvk.dts    | 29 ++++++
 .../arm/boot/dts/qcom-ipq4019-hap-ac3.dts     | 32 +++++++
 .../arm/boot/dts/qcom-ipq4019-lhgg-60ad.dts   | 18 ++++
 .../arm/boot/dts/qcom-ipq4019-map-ac2200.dts  | 21 +++++
 .../arch/arm/boot/dts/qcom-ipq4019-mf286d.dts | 36 +++++++
 .../arm/boot/dts/qcom-ipq4019-r619ac.dtsi     | 36 +++++++
 .../arm/boot/dts/qcom-ipq4019-rtl30vw.dts     | 20 ++++
 .../arm/boot/dts/qcom-ipq4019-xx8300.dtsi     | 30 +++++-
 .../arch/arm/boot/dts/qcom-ipq4028-wpj428.dts | 32 +++++++
 .../dts/qcom-ipq4029-aruba-glenmorangie.dtsi  | 20 +++-
 .../arm/boot/dts/qcom-ipq4029-gl-b1300.dts    | 24 +++++
 .../boot/dts/qcom-ipq4029-insect-common.dtsi  | 38 ++++++++
 target/linux/ipq40xx/image/Makefile           |  2 +
 32 files changed, 754 insertions(+), 99 deletions(-)
 create mode 100644 target/linux/ipq40xx/base-files/etc/board.d/05_compat-version
 delete mode 100644 target/linux/ipq40xx/base-files/lib/preinit/06_set_preinit_iface_ipq40xx.sh

diff --git a/target/linux/ipq40xx/base-files/etc/board.d/01_leds b/target/linux/ipq40xx/base-files/etc/board.d/01_leds
index 884b265dcb62a..566893769f146 100644
