--- a/target/linux/ipq40xx/base-files/etc/board.d/02_network
+++ b/target/linux/ipq40xx/base-files/etc/board.d/02_network
@@ -100,10 +102,8 @@ ipq40xx_setup_interfaces()
 		ucidef_add_switch "switch0" \
 			"0u@eth0" "2:lan:3" "3:lan:2" "4:lan:1"
 		;;
-	cellc,rtl30vw)
-		ucidef_set_interface_lan "eth0"
-		ucidef_add_switch "switch0" \
-			"0u@eth0" "3:lan" "4:lan"
+	compex,wpj428)
+		ucidef_set_interface_lan "lan1 lan2"
 		;;
 	devolo,magic-2-wifi-next)
 		ucidef_set_interface_lan "eth0 eth1 eth2"
