--- a/target/linux/ipq40xx/files/drivers/net/dsa/qca/qca8k-ipq4019.c
+++ b/target/linux/ipq40xx/files/drivers/net/dsa/qca/qca8k-ipq4019.c
@@ -595,8 +448,11 @@ qca8k_port_set_status(struct qca8k_priv *priv, int port, int enable)
 {
 	u32 mask = QCA8K_PORT_STATUS_TXMAC | QCA8K_PORT_STATUS_RXMAC;
 
-	/* Port 0 and 6 have no internal PHY */
-	if (port > 0 && port < 6)
+	/* Port 0 is internally connected to the CPU
+	 * TODO: Probably check for RGMII as well if it doesnt work
+	 * in RGMII mode.
+	 */
+	if (port > QCA8K_CPU_PORT)
 		mask |= QCA8K_PORT_STATUS_LINK_AUTO;
 
 	if (enable)
